rclpy/src/rclpy/node.cpp

```diff
@@ -140,19 +144,36 @@ get_node_names_impl(py::capsule pynode, bool get_enclaves)
     }
   }
 
-  rcutils_ret_t rcutils_ret;
-  rcutils_ret = rcutils_string_array_fini(&node_names);
-  if (RCUTILS_RET_OK != rcutils_ret) {
-    throw RCLError("Failed to destroy node names");
-  }
-  rcutils_ret = rcutils_string_array_fini(&node_namespaces);
-  if (RCUTILS_RET_OK != rcutils_ret) {
-    throw RCLError("Failed to destroy node_namespaces");
-  }
-  rcutils_ret = rcutils_string_array_fini(&enclaves);
-  if (RCUTILS_RET_OK != rcutils_ret) {
-    throw RCLError("Failed to destroy enclaves string array");
-  }
+  RCPPUTILS_SCOPE_EXIT(
```

Comment:
@gbalke this block should go before populating the list, around line 128. [`RCPPUTILS_SCOPE_EXIT`](https://github.com/ros2/rcpputils/blob/master/include/rcpputils/scope_exit.hpp) block will be executed upon leaving the scope it was defined in.